### PR TITLE
params: show only params files that are actually used

### LINKED ISSUE #5539: Inconsistiency in `params.yaml`
While working on #5451 I stumbled upon an inconsistency in our tests.

If we take a look at params test from [here](https://github.com/iterative/dvc/blob/c3a0e8733e799e3c49e5ab6b9a8e32f365553f97/tests/unit/test_params.py#L6):
```python
def test_params_order(tmp_dir, dvc):
    tmp_dir.gen(
        {
            "params.yaml": dumps_yaml({"p1": 1}),
            "p1.yaml": dumps_yaml({"p2": 1}),
            "sub": {"p2.yaml": dumps_yaml({"p3": 1})},
        }
    )

    p2_path = os.path.join("sub", "p2.yaml")
    dvc.stage.add(
        params=[{p2_path: ["p3"]}, {"p1.yaml": ["p2"]}],
        cmd="cmd1",
        name="stage1",
    )
    with (tmp_dir / "sub").chdir():
        dvc.stage.add(params=["p1"], cmd="cmd2", name="stage2")

    # params are sorted during dumping, therefore p1 is first
    assert list(dvc.params.show()[""]) == ["p1.yaml", p2_path, "params.yaml"]
```

We expect that providing `p1` param while in `subdir` will resolve into `{repo_root}/params.yaml:foo`

On the other hand, one of the tests for [experiments](https://github.com/iterative/dvc/blob/c3a0e8733e799e3c49e5ab6b9a8e32f365553f97/tests/func/experiments/test_experiments.py#L490):
```python
def test_subdir(tmp_dir, scm, dvc, workspace):
    subdir = tmp_dir / "dir"
    subdir.gen("copy.py", COPY_SCRIPT)
    subdir.gen("params.yaml", "foo: 1")

    with subdir.chdir():
        dvc.run(
            cmd="python copy.py params.yaml metrics.yaml",
            metrics_no_cache=["metrics.yaml"],
            params=["foo"],
            name="copy-file",
            no_exec=True,
        )
        scm.add(
            [subdir / "dvc.yaml", subdir / "copy.py", subdir / "params.yaml"]
        )
        scm.commit("init")

        results = dvc.experiments.run(
            PIPELINE_FILE, params=["foo=2"], tmp_dir=not workspace
        )
        assert results

    exp = first(results)
    ref_info = first(exp_refs_by_rev(scm, exp))

    fs = scm.get_fs(exp)
    for fname in ["metrics.yaml", "dvc.lock"]:
        assert fs.exists(subdir / fname)
    with fs.open(subdir / "metrics.yaml") as fobj:
        assert fobj.read().strip() == "foo: 2"

    assert dvc.experiments.get_exact_name(exp) == ref_info.name
    assert scm.resolve_rev(ref_info.name) == exp
```
In this case `foo` resolves to `{repo_root}/subdir/params.yaml`

It would be possible that both use cases to be correct, if we expect some kind of "fallback" (if not `params.yaml` in `cwd`, try in root, or previous dir) but that looks to me like bug, because we expose ourselves to risk that introducing `params.yaml` in some subdirectory will change the source of parameters, which will result in indirect influence on some pipeline just by creating file with certain param name inside.

So, I wanted to raise a discussion, whether we assume that there is only single default `{repo_root}/params.yaml`? Or, when we are specifying `--params foo` we expect `{cwd}/params.yaml` to exist and contain `foo`?

### PR DESCRIPTION
**Pull request by pared**
* [x] ❗ I have followed the [Contributing to DVC](https://dvc.org/doc/user-guide/contributing/core) checklist.

* [x] 📖 If this PR requires [documentation](https://dvc.org/doc) updates, I have created a separate PR (or issue, at least) in [dvc.org](https://github.com/iterative/dvc.org) and linked it here.

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

Closes #5539

### CODE REVIEW

File: tests/unit/test_params.py [hunk 1]
```diff
@@ -22,7 +25,11 @@ def test_params_order(tmp_dir, dvc):
         dvc.stage.add(params=["p1"], cmd="cmd2", name="stage2")
```
Comment (skshetry) [inline tests/unit/test_params.py:25]: I wonder why this didn't fail before.

Comment (skshetry) [reply tests/unit/test_params.py:25]: Ahh, because it does not check if the dependencies exist.

**Review by skshetry — APPROVED** [review]
Looks good to me. But this default behaviour has been there since we introduced `params`, same as #5451 (introduced in #3631), so cc'ing @efiop if this was intended.

**Comment by efiop** [comment]
It was intended originally, but now it is a mixup. Need to take a closer look and also look at the docs, as those might be affected. IIRC the logic was that people might use it without our pipelines (similar to same assumptions/discussions around metrics/plots). I don't feel comfortable just disabling this behavior, because it might turn out that it is needed in some scenario. We might need to double check params workflow first.

**Comment by skshetry** [comment]
@efiop, isn't expecting a `params.yaml` file at the root of a git repository quite an assumption?

**Comment by efiop** [comment]
@skshetry Hard to tell. Params/metrics and plots are all filled with assumptions that were provided by users while developing it, and I also get constantly confused by it. The change seems logical, but we just need to make sure it fits into the workflow and we def need to check the docs.

**Comment by pared** [comment]
@efiop

So the whole issue started in https://github.com/iterative/dvc/issues/5451, where user has quite extensive pipeline, and running `exp show` yielded unwanted results. This change was a prerequisite to make sure only used parameters are shown. 

> Params/metrics and plots are all filled with assumptions that were provided by users while developing it, and I also get constantly confused by it.

I guess we all are. 
I think I jumped the horse here. The way to fix #5451 and not break just released 2.0.0 is to provide new flag that will allow user to show only used params, and keep this issue in mind when releasing 3.0. That way we keep the default behaviour.

**Comment by pared** [comment]
Closing, we do not intend to change default behaviour for now.
Context: 
https://github.com/iterative/dvc/issues/5451#issuecomment-793992639

**Pull request closed without merging**